docs/source/app_developers_guide/docker.rst

```diff
@@ -0,0 +1,416 @@
+.. note::
+
+  Genesis block, validator and transaction processor startup are handled for you
+  if you're using the Docker workflow. If you want a better understanding of
+  this process, see :doc:`ubuntu`.
+
+
+Confirming Connectivity 
+=======================
+
+To confirm that a validator is up and running, and reachable from the client
+container, you can use the curl command:
+
+.. code-block:: console
+
+	root@75b380886502:/# curl http://rest_api:8080/blocks
+
+If the validator is running and reachable, the output should be similar to:
+
+.. code-block:: console
+
+	{
+	  "data": [
+	    {
+	      "batches": [],
+	      "header": {
+	        "batch_ids": [],
+	        "block_num": 0,
+	        "consensus": "R2VuZXNpcw==",
+	        "previous_block_id": "0000000000000000",
+	        "signer_pubkey": "03061436bef428626d11c17782f9e9bd8bea55ce767eb7349f633d4bfea4dd4ae9",
+	        "state_root_hash": "708ca7fbb701799bb387f2e50deaca402e8502abe229f705693d2d4f350e1ad6"
+	      },
+	      "header_signature": "119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b"
+	    }
+	  ],
+	  "head": "119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b",
+	  "link": "http://rest_api:8080/blocks?head=119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b",
+	  "paging": {
+	    "start_index": 0,
+	    "total_count": 1
+	  }
+	}root@75b380886502:/#
+
+If the validator process, or the validator container, is not running, the curl
+command will return nothing, or time out.
+
+
+Transaction Processors
+======================
+
+The Docker Compose file (*sawtooth-default.yaml*) starts the following
+transaction processors automatically:
+
+* tp_config
+* tp_intkey_python
+* tp_xo_python
+
+These transaction processors handle transactions for the config, intkey, and XO transaction families, and are written in Python.
+
+
+Creating And Submitting Transactions
+====================================
+
+The **intkey** command is provided to create sample transactions of the intkey
+transaction type for testing purposes.
+
+This section guide you through the following tasks:
+
+1. Prepare a batch of intkey transactions that set the keys to random values.
+
+2. Generate *inc* (increment) and *dec* (decrement) transactions to apply to
+   the existing state stored in the blockchain.
+
+3. Submit these transactions to the validator.
+
+Run the following commands from the client container:
+
+.. code-block:: console
+
+  $ intkey create_batch
+  $ intkey load -f batches.intkey -U http://rest_api:8080
+
+You can observe the processing of the intkey transactions by observing the
+logging output of the intkey transaction processor and the validator, from the
+terminal window in which you ran the *docker-compose* command. A truncated
+example of the output:
```

Comment:
The terminal window in which you ran the *docker-compose* command will begin logging output as the validator and intkey transaction processor handle the transactions just submitted.